src/lib/Configuration/AggregateProvider.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformRichText\Configuration;
+
+use EzSystems\EzPlatformRichText\API\Configuration\ProviderService;
+
+/**
+ * RichText configuration provider, providing configuration by aggregating different sources.
+ *
+ * @internal type-hint \EzSystems\EzPlatformRichText\API\Configuration\ProviderService
+ */
+final class AggregateProvider implements ProviderService
```

Comment:
Unit tests are missing for this class.

Reply:
Added via 00f92787c.

I've also added missing coverage for configuration providers which are passed to that aggregate. However I'm not really sure about them. It's mock on top of mock to describe control flow instead of something more essential (see 47a66d6e7 and 99bf08ed2). We need here some Symfony full stack end-to-end test which for given semantic config will return expected UI configuration.

Creating those tests forced me to create common interface for `CustomStyle` and `CustomTag` providers to mock them, which, given the nature of them - stored as custom Templates - is probably a good thing... well, maybe except for auto-wiring no longer working. See d8cb3e0eb.